src/posit/connect/jobs.py

```diff
@@ -99,13 +100,14 @@ class _Job(TypedDict):
         tag: Required[JobTag]
         """A tag categorizing the job type. Options are build_jupyter, build_report, build_site, configure_report, git, packrat_restore, python_restore, render_shiny, run_api, run_app, run_bokeh_app, run_dash_app, run_fastapi_app, run_pyshiny_app, run_python_api, run_streamlit, run_tensorflow, run_voila_app, testing, unknown, val_py_ext_pkg, val_r_ext_pkg, and val_r_install."""
 
-    def __init__(self, ctx, parent: Active, **kwargs: Unpack[_Job]):
-        super().__init__(ctx, parent, **kwargs)
-        self._parent = parent
+    @overload
+    def __init__(self, ctx: Context, path: str, pathinfo: str, /, **attributes: Unpack[_Job]): ...
+
+    @overload
+    def __init__(self, ctx: Context, path: str, pathinfo: str, /, **attributes: Any): ...
```

Comment:
Its too bad these overloads are necessary

Reply:
They've grown on me. A large part of this project is documentation. Even more so now that we are abstracting out the API calls.

Reply:
From out conversation...

Fallback overload was added to avoid breaking types when users use new API values that are not implemented.

While it works, it breaks type checking for all other variables. I'm in the stance of "If the user chose to add the new parameter. Then they can ignore the type checking for the parameter."

So I would ask to remove the fallback overload.